raiden/smart_contracts/NettingChannelLibrary.sol

```diff
@@ -282,15 +295,47 @@ library NettingChannelLibrary {
             throw;
         }
 
+        uint64 nonce;
         assembly {
-            nonce := mload(add(transfer_raw, 12))  // skip cmdid and padding
+            nonce := mload(add(transfer, 12))  // skip cmdid and padding
         }
-
         if (nonce < sender.nonce || nonce == sender.nonce) {
             throw;
         }
-
         decodeAndAssign(sender, transfer_raw);
+    }
+
+    /// @notice updateTransfer Updates (disputes) the state after closing.
+    /// @param caller_address The counterparty to the channel. The participant
+    ///                       that did not close the channel.
+    /// @param first_transfer The transfer the counterparty believes is the
+    ///                        valid state for the first participant.
+    /// @param second_transfer The transfer the counterparty believes is the
+    ///                        valid state for the second participant.
```

Comment:
To do that (and I think it's better from a logic point of view), other functions in the contract should also change. In `close()` function of the same contract it has the same kind of logic:

```
    /// @param first_encoded the last transfer of the first participant
    /// @param second_encoded the last transfer of the second participant
```

And there it does not matter whose the transaction is, first and second transfers can be given in any order.